app/models/user.rb

```diff
@@ -1,8 +1,12 @@
 class User < ActiveRecord::Base
   authenticates_with_sorcery!
 
-  validates :email, :password, presence: true
+  validates :email, presence: true
   validates :email, uniqueness: true, if: -> { email.present? }
+  validates :password, presence: true, on: :update, confirmation: true
+
+  before_create :setup_activation
```

Comment:
What does `setup_activation` do? Since its not part of this model? Is that something Sorcery provides?